### ISSUE TITLE
Bug generating "Forecast Report"

### ISSUE THREAD

**Issue by RJPC** [adempiere/adempiere#1620]
El reporte se genera en blanco, adjunto imagenes y log.

**Parameter**
![parametros](https://user-images.githubusercontent.com/5696550/37933387-8a868e1a-313a-11e8-952c-d5265625de41.png)

**Report**
![reporte](https://user-images.githubusercontent.com/5696550/37933396-9267d904-313a-11e8-8342-255716861555.png)

**Log**
[trace.log](https://github.com/adempiere/adempiere/files/1849557/trace.1.log)

**Comment by marcalwestf** [comment]
Trying to reproduce the error, I open the report _Manufacturing Management/Planning Management/Forecast Management/ Forecast Report_ on a clean Adempiere390 database.
It displays the same parameters as your first image. I then run the report without filling a single parameter.

However, I do get results running the report:
![20180326_issue_1620](https://user-images.githubusercontent.com/1789408/37941194-3408f556-312a-11e8-9056-53e016120317.png)

I get an error in the console 

> -----------> MReportView.get_Value: Column not found - Classname 

which maybe worth looking into, but for now seem not to be relevant.

So, I cannot generate an error.

Looking at the code changes, I see that the view _RV_M_Forecast_ has been improved: instead of repeating joins three times in columns, they are moved to the main join. Which is OK.

I would say it is an improvement, but I wouldn't label it as a bug.

Now, I am going to test the changes and report the results.

**Comment by yamelsenih** [comment]
Hi @marcalwestf it error is when you have more than one price list version for a same product, in garden just exists one price list version by price list

**Comment by marcalwestf** [comment]
.... ran the XML _03780_1620_Fixed_RV_M_Forecast_View.xml_ (in Eclipse and Intellij).
It delivers the following error:
![20180326_issue_1620_2](https://user-images.githubusercontent.com/1789408/37941679-a0ca79a6-312c-11e8-9d23-5437eb1ed583.png)

Also in the console the following message:
> 19:21:08.727     PgConnection.setAutoCommit:   setAutoCommit = {0} (true) [85]
> 19:21:08.727           Trx.close: POSave_988fd409-182e-45ae-ba77-65417ae2f1b8 [85]
> 19:21:08.727     MigrationFromXML.unlock: MPInstance[1000492,OK=falsejava.lang.NullPointerException] [85]
> 19:21:08.747       ProcessCtl.run: unlock - java.lang.NullPointerException [15]

I don't know if it is required to have other XMLs executed.

I am going to run the SQL in the XML, to see if the error stems from it.....

**Comment by marcalwestf** [comment]
> Hi @marcalwestf it error is when you have more than one price list version for a same product, in garden just exists one price list version by price list
Thanks for explaining why I could not reproduce; I didn't know.

**Comment by marcalwestf** [comment]
well, it seems that it is required to have run the _XML 03560_1521_FileNameFilePathOrName.xml_  which corrects file name to _FilePathOrName_ in order to avoid the NPE.

Once run this XML, the import executes without errors, and the migration runs also well.

I then ran the report  open the report _Manufacturing Management/Planning Management/Forecast Management/ Forecast Report_  and it delivers results.

I found a small bug concerning zooming to _Operational Period_ in embedded tab _Line_ of window _Forecast_ : it is not possible to zoom. But I will create an  issue this regarding:
![20180326_issue_1620_3](https://user-images.githubusercontent.com/1789408/37983865-e55d1af8-31b1-11e8-92d5-88d3ee54f453.png)
